aot-maven-plugin/src/main/java/com/dylibso/chicory/maven/aot/AotGenMojo.java

```diff
@@ -0,0 +1,137 @@
+package com.dylibso.chicory.maven.aot;
+
+import static com.github.javaparser.StaticJavaParser.parseClassOrInterfaceType;
+import static com.github.javaparser.StaticJavaParser.parseType;
+
+import com.dylibso.chicory.aot.AotMachine;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.runtime.Machine;
+import com.dylibso.chicory.wasm.Parser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.ObjectCreationExpr;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.utils.SourceRoot;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * This plugin generates an invokable library from the compiled Wasm
+ */
+@Mojo(name = "wasm-aot-gen", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = false)
+public class AotGenMojo extends AbstractMojo {
+
+    /**
+     * the wasm module to be used
+     */
+    @Parameter(required = true)
+    private File wasmFile;
+
+    /**
+     * the name to be used by the generated class
+     */
+    @Parameter(required = true)
+    private String name;
+
+    /**
+     * the target folder to generate classes
+     */
+    @Parameter(
+            required = true,
+            defaultValue = "${project.basedir}/target/generated-resources/chicory-aot")
+    private File targetClassFolder;
+
+    /**
+     * the target source folder to generate the Machine implementation
+     */
+    @Parameter(
+            required = true,
+            defaultValue = "${project.basedir}/target/generated-sources/chicory-aot")
+    private File targetSourceFolder;
+
+    /**
+     * The current Maven project.
+     */
+    @Parameter(property = "project", required = true, readonly = true)
+    private MavenProject project;
+
+    @Override
+    @SuppressWarnings({"StringSplitter", "deprecation"})
+    public void execute() throws MojoExecutionException {
+        var instance =
+                Instance.builder(Parser.parse(wasmFile))
+                        .withMachineFactory(inst -> new AotMachine(name, inst))
+                        .withStart(false)
+                        .withInitialize(false)
+                        .withSkipImportMapping(true)
+                        .build();
+        var compiled = ((AotMachine) instance.getMachine()).compiledClass();
```

Comment:
With #565, this can become
```java
var module = Parser.parse(wasmFile);
var compiler = new AotCompiler(module, name);
var compiled = compiler.compileClassToBytes();
// verify that compiled class is valid
compiler.loadClassBytes(compiled);

Reply:
Sounds like an ideal API, just give me a little to review 565 :+1: thanks a lot for the support!

Reply:
I updated the API to allow returning multiple classes, which is needed for inlining `AotMethods`, method splitting, etc. With that change, this code can become something like
```java
for (Entry<String, byte[]> entry : result.classBytes()) {
    var targetFile = targetClassFolder.toPath().resolve(entry.getKey() + ".class");
    try {
        Files.write(targetFile, entry.getValue());
    } catch (IOException e) {
        throw new MojoExecutionException("Failed to write " + targetFile, e);
    }
}